java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/SearchService.java

```diff
@@ -157,6 +164,38 @@ public PoiSearchResult searchPoi(SearchService.PoiSearchData data) throws IOExce
         }
     }
     
+    public Feature searchPoiByLatlon(LatLon loc, long osmid) throws IOException {
+        List<LatLon> bbox = new ArrayList<>();
+        bbox.add(new LatLon(loc.getLatitude() + 0.0015, loc.getLongitude() - 0.0015));
+        bbox.add(new LatLon(loc.getLatitude() - 0.0015, loc.getLongitude() + 0.0015));
+        QuadRect searchBbox = getSearchBbox(bbox);
+        
+        List<BinaryMapIndexReader> usedMapList = new ArrayList<>();
+        SearchResult res = null;
+        try {
+            List<OsmAndMapsService.BinaryMapIndexReaderReference> mapList = getMapsForSearch(bbox, searchBbox);
+            if (!mapList.isEmpty()) {
+                usedMapList = osmAndMapsService.getReaders(mapList, null);
+                SearchUICore.SearchResultCollection resultCollection = searchAllPoiByBbox(searchBbox, usedMapList);
+                if (resultCollection != null) {
+                    List<SearchResult> searchResults = resultCollection.getCurrentSearchResults();
+                    if (!searchResults.isEmpty()) {
+                        res = searchResults.stream()
```

Comment:
1. No need to store all results to retrieve 1 object. Probably better way should be used via accept / filter
2. No need to use search at all as we just need to read POI object